lib/engine/phase.rb

```diff
@@ -31,6 +31,10 @@ def current
       @phases[@index]
     end
 
+    def train_limit(entity)
```

Comment:
entity = nil

Reply:
and handle nil entity

Reply:
i see i guess you enforce it to exist, i guess it's ok

Reply:
I think it is fine to not have default value here, and instead enforce it to be specified. Did update all calls, and they all had access to a corporation.